kong/plugins/ip-restriction/handler.lua

```diff
@@ -29,6 +31,31 @@ do
 end
 
 
+local is_http_subsystem = ngx.config.subsystem == "http"
+
+
+local do_exit
+if is_http_subsystem then
+  do_exit = function(status, message)
+    return kong.response.error(status, message)
+  end
+else
+  do_exit = function(status, message)
+    local cjson_encode = cjson.encode
+    local tcpsock, err = ngx_req.socket(true)
+    if err then
+      error(err)
+    end
+
+    tcpsock:send(cjson_encode({
+      status  = status,
+      message = message
+    }))
+
+    return ngx.exit()
```

Comment:
1. It is better to localize `ngx.exit`

```lua
local ngx_exit = ngx.exit

...

ngx_exit()
```

2. we should use `ngx.exit()` with a status argument, e.g. `ngx.exit(1)`